### DISCUSSION TITLE
Integrating with rails application

### DISCUSSION THREAD

**Discussion by mohro** [facebook/docusaurus#8747] [Q&A]
Hi, 
 I have a rails application on the '/' path. We use nginx for reverse proxy and I am planning to deploy docusaurus build using it.
 Is it possible to have single docusaurus instance handle only the `/blog` and `/docs` paths and skip the routes managed by the rails application?

**Comment by slorber** [comment]
Similar question to https://github.com/facebook/docusaurus/discussions/8728

I'd recommend putting the Docusaurus site under a baseUrl instead of trying to merge it with Rails. Technically it's possible but you might encounter conflicts so it might be better to isolate each app better.

**Reply by mohro** [reply]
Thanks and I will come back if I have more questions.

**Comment by marvindanig** [comment]
hi @slorber 

I came across the following [note](https://github.com/facebook/docusaurus/discussions/7973#discussion-4314553 ) on a different but somewhat related thread. Paraphrasing the set-up by @nickbabcock for a quick reference: 

```
/ <----------- Handled by rails.
├── /account
├── /api        
├── /changelog
# ...
# ...
├── /blog <--- Handled by docusaurus without prepending some `/baseUrl/` string.
├── /docs 
```

Can we colocate a single docusaurus instance alongside the rails app without using a prepend `baseUrl` (like `/content/`) in between to serve both `/blog` and `/docs` from under the root directory? Considering the discussion at [#8728](https://github.com/facebook/docusaurus/discussions/8728), aside from the issue of `assets` folder naming conflict, I think this type of setup appears to be quite an ask. Or am I thinking about it the wrong way?

Edited: `baseURL` → `baseUrl`.

**Reply by slorber** [reply]
You can colocate whatever you want with whatever you want but it's your responsibility to manage conflicts if you decide to do so, we can't guarantee a seamless integration with any third-party technology out there because routes and file paths might conflict.

Using a baseurl is a good security to ensure conflicts to not happen.

Now if you really want to do that, I guess it shouldn't be a problem with Rails but you could use a different path for the Rails asset pipeline. https://guides.rubyonrails.org/asset_pipeline.html#changing-the-assets-path
(I don't have a lot of experience with Rails but this should work 🤷‍♂️ )
